packages/scaffold-ui/src/partials/w3m-social-login-widget/index.ts

```diff
@@ -153,37 +154,57 @@ export class W3mSocialLoginWidget extends LitElement {
   async onSocialClick(socialProvider?: SocialProvider) {
     if (socialProvider) {
       AccountController.setSocialProvider(socialProvider, ChainController.state.activeChain)
+
       EventsController.sendEvent({
         type: 'track',
         event: 'SOCIAL_LOGIN_STARTED',
         properties: { provider: socialProvider }
       })
-      RouterController.push('ConnectingSocial')
     }
-    const authConnector = ConnectorController.getAuthConnector()
-    this.popupWindow = CoreHelperUtil.returnOpenHref(
-      '',
-      'popupWindow',
-      'width=600,height=800,scrollbars=yes'
-    )
+    if (socialProvider === SocialProviderEnum.Farcaster) {
+      RouterController.push('ConnectingFarcaster')
+      const authConnector = ConnectorController.getAuthConnector()
+
+      if (authConnector) {
+        if (!AccountController.state.farcasterUrl) {
+          try {
+            const { url } = await authConnector.provider.getFarcasterUri()
```

Comment:
this is what we maybe can call earlier in the flow and have the uri available? just spitballing

Reply:
I tried this, but magic/farcaster are calling status calls after we request the farcaster URI.